helix-core/src/main/java/org/apache/helix/monitoring/mbeans/InstanceMonitor.java

```diff
@@ -24,35 +24,86 @@
 import java.util.Map;
 import java.util.Set;
 
+import javax.management.JMException;
+import javax.management.ObjectName;
+
 import com.google.common.base.Joiner;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Lists;
+import org.apache.helix.monitoring.mbeans.dynamicMBeans.DynamicMBeanProvider;
+import org.apache.helix.monitoring.mbeans.dynamicMBeans.DynamicMetric;
+import org.apache.helix.monitoring.mbeans.dynamicMBeans.SimpleDynamicMetric;
+
 
 /**
  * Implementation of the instance status bean
  */
-public class InstanceMonitor implements InstanceMonitorMBean {
+public class InstanceMonitor extends DynamicMBeanProvider {
+  /**
+   * Metric names for instance capacity.
+   */
+  public enum InstanceMonitorMetrics {
+    TOTAL_MESSAGE_RECEIVED_COUNTER("TotalMessageReceivedCounter"),
+    ENABLED_STATUS_GAUGE("EnabledStatusGauge"),
+    ONLINE_STATUS_GAUGE("OnlineStatusGauge"),
+    MAX_CAPACITY_USAGE_GAUGE("MaxCapacityUsageGauge"),
+    DISABLED_PARTITIONS_GAUGE("DisabledPartitionsGauge");
```

Comment:
Don't change the name of the existing metrics. Or this PR will break the existing usage of those metrics.

Reply:
That is not right, previously the metric names were directly derived from the method name. Your change will create new names and disappear the old ones.

Reply:
Changed. I am cleared about the static metric names. The names are decided by the getters' names.